### PR TITLE
Keyword `$vocabulary`

### PR DESCRIPTION
**Pull request by AgniveshChaubey**
This PR addresses #52.

I am not very confident with the last three keywords, so will require a bit more guidance on these keywords.

### CODE REVIEW

File: content/2020-12/core/vocabulary.markdown [hunk 1]
```diff
@@ -18,3 +18,83 @@ related:
   - vocabulary: core
     keyword: $schema
 ---
+
+Vocabularies are sets of keywords and their meanings used to validate JSON documents. This keyword is crucial for meta-schemas, which are schemas that define other schemas.
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:22]: I think the first sentence is a bit confusing. A vocabulary is an atomic collection of interrelated keywords that can be used in a JSON Schema. These keywords are not only for validation. i.e. you have previously documented annotation keywords like `title`, that come from the "Meta Data" vocabulary

File: content/2020-12/core/vocabulary.markdown [hunk 2]
```diff
@@ -18,3 +18,83 @@ related:
...
+
+The `$vocabulary` keyword is used in meta-schemas to identify the vocabularies available for use in schemas described by that meta-schema. It is also used to indicate whether each vocabulary is required or optional, in the sense that an implementation must understand the required vocabularies in order to successfully process the schema.
+
+* **Required and optional vocabularies:** If a vocabulary is required and an implementation does not recognize it, it must refuse to process any schemas that declare this meta-schema. If a vocabulary is optional, implementations that do not recognize it should proceed with processing such schemas.
+
+* **Mandatory:** The core vocabulary MUST always be included and set as required.
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:28]:
```suggestion
* **Mandatory:** The Core vocabulary MUST always be included and set as required.
```

File: content/2020-12/core/vocabulary.markdown [hunk 3]
```diff
@@ -18,3 +18,83 @@ related:
...
+
+* **Non-inheritability:** Vocabularies defined in one meta-schema do not automatically apply to another meta-schema that references it. Each meta-schema must declare its vocabularies independently.
+
+* **Recommendation:** Meta-schemas should always declare this keyword to clearly specify the vocabularies in use and avoid ambiguities.
+
+## Examples
+
+{{<schema `'$vocabulary' for default meta-schema`>}}
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:36]:
```suggestion
{{<schema `'$vocabulary' for default official 2020-12 meta-schema`>}}
```

File: content/2020-12/core/vocabulary.markdown [hunk 4]
```diff
@@ -18,3 +18,83 @@ related:
...
+  ],
+  // ...
+}
+{{</schema>}}
+
+{{<schema `Custom vocabulary with a custom keyword`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://example.com/meta/example-vocab",
+  "$dynamicAnchor": "meta",
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:66]: I think you should explain `$dynamicAnchor: meta` in this page. You can explain it is set by convention to `meta` on the official meta-schemas, and it's a mechanism to enable meta-schema extensibility. By declaring `$dynamicAnchor: meta` here, you are essentially telling JSON Schema to validate EVERY subschema of the instance schema against the meta-schema, and not only the top level of it

File: content/2020-12/core/vocabulary.markdown [hunk 5]
```diff
@@ -18,3 +18,83 @@ related:
...
+  "type": [ "object", "boolean" ],
+  "properties": {
+    "minDate": {
+      "type": "string",
+      "pattern": "\d\d\d\d-\d\d-\d\d",
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:71]: This pattern is invalid JSON. You have to escape the backslashes

File: content/2020-12/core/vocabulary.markdown [hunk 6]
```diff
@@ -18,3 +18,83 @@ related:
...
+    { "$ref": "meta/validation" },
+    { "$ref": "meta/meta-data" },
+    { "$ref": "meta/format-annotation" },
+    { "$ref": "meta/content" }
+  ],
+  // ...
+}
+{{</schema>}}
+
+{{<schema `Custom vocabulary with a custom keyword`>}}
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:62]: This is not a custom vocabulary, but a vocabulary meta-schema

File: content/2020-12/core/vocabulary.markdown [hunk 7]
```diff
@@ -18,3 +18,83 @@ related:
...
+    "minDate": {
+      "type": "string",
+      "pattern": "\d\d\d\d-\d\d-\d\d",
+      "format": "date"
+    }
+  }
+}
+{{</schema>}}
+
+{{<schema `Meta-schema that includes the above vocabulary`>}}
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:78]: Maybe add a variant of this example too importing the example vocabulary as optional? Just to exemplify that

**Comment by jviotti** [comment]
I left a few comments but definitely looks in the right direction. I think you mostly get it already

File: content/2020-12/core/vocabulary.markdown (hunk 1 above)
Comment (AgniveshChaubey) [reply content/2020-12/core/vocabulary.markdown:22]: Okay, lets omit that

File: content/2020-12/core/vocabulary.markdown (hunk 6 above)
Comment (AgniveshChaubey) [reply content/2020-12/core/vocabulary.markdown:62]: can you provide an example of a custom vocabulary?

File: content/2020-12/core/vocabulary.markdown (hunk 4 above)
Comment (AgniveshChaubey) [reply content/2020-12/core/vocabulary.markdown:66]: Should I add it under <learning-more> block or just below that schema block as a note?

Comment (AgniveshChaubey) [reply content/2020-12/core/vocabulary.markdown:66]: Should I add it under `leaning-more` block or just below that schema block as a note?

Comment (jviotti) [reply content/2020-12/core/vocabulary.markdown:64]: Yeah, that's a good point. Let's use the new `learning-more` block :)

File: content/2020-12/core/vocabulary.markdown [hunk 8]
```diff
@@ -18,3 +18,99 @@ related:
...
+    "https://example.com/vocab/example-vocab": "true"
+  },
+  "allOf": [
+    { "$ref": "https://json-schema.org/draft/2020-12/meta/core" },
+    { "$ref": "https://example.com/meta/example-vocab" }
+  ]
+}
+{{</schema>}}
+
+{{<schema `Meta-schema with the above vocabularyas optional`>}}
```
Comment (jviotti) [inline content/2020-12/core/vocabulary.markdown:94]:
```suggestion
{{<schema `Meta-schema with the above vocabulary as optional`>}}
```

**Pull request merged**